errbot/repo_manager.py

```diff
@@ -104,7 +104,8 @@ def shutdown(self):
     def check_for_index_update(self):
         if REPO_INDEX not in self:
             log.info('No repo index, creating it.')
-            self[REPO_INDEX] = {LAST_UPDATE: 0}
+            self.index_update()
+            return
```

Comment:
I added an early return here, otherwise it could happen that `self.index_update()` fails and causes the check below to fire it again. Otherwise, looks good. I'll merge this in, thanks!